################################################################################
SHORT DESCRIPTION:
################################################################################
Do not add tty0 to the preferred consoles.

################################################################################
LONG DESCRIPTION:
################################################################################
Preferred consoles behavior used to not affect dom0 until:
47b02f4c621c x86/xen: add tty0 and hvc0 as preferred consoles for dom0
71dc05635983 x86/Xen: further refine add_preferred_console() invocations

Since these changes, PV guests will always output on tty0 and hvc0 regardless
of console= parameter. This makes some noise at boot.

None of the PV kernels used in OpenXT rely on that, yet silencing the loglevel
is not desired as the serial output is often used to debug boot issues. This
change temporarily does _not_ add tty0 as a default preferred console.

console_set_on_cmdline is not usable at this stage unfortunately since the
cmdline has not been parsed yet.

################################################################################
REMOVAL
################################################################################
Once a better alternative is available, this patch should be removed.

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
None.

################################################################################
PATCHES
################################################################################
--- a/arch/x86/xen/enlighten_pv.c
+++ b/arch/x86/xen/enlighten_pv.c
@@ -1397,11 +1397,7 @@ asmlinkage __visible void __init xen_sta
 		xen_boot_params_init_edd();
 	}
 
-	if (!boot_params.screen_info.orig_video_isVGA)
-		add_preferred_console("tty", 0, NULL);
 	add_preferred_console("hvc", 0, NULL);
-	if (boot_params.screen_info.orig_video_isVGA)
-		add_preferred_console("tty", 0, NULL);
 
 #ifdef CONFIG_PCI
 	/* PCI BIOS service won't work from a PV guest. */
